flink-runtime/src/main/java/org/apache/flink/runtime/webmonitor/WebMonitorUtils.java

```diff
@@ -207,6 +215,53 @@ public static WebMonitor startWebRuntimeMonitor(
 		}
 	}
 
+	@SuppressWarnings({"unchecked", "rawtypes"})
+	public static <T extends RestfulGateway> List<Tuple2<RestHandlerSpecification, ChannelInboundHandler>> tryLoadJarUploadHandler(
```

Comment:
Should be `T extends DispatcherGateway`

Reply:
Could return a `Collection` instead of `List`

Reply:
both comments fixed